aci/data_source_aci_fvrsprovcons.go

```diff
@@ -17,110 +17,91 @@ func dataSourceAciContractProvider() *schema.Resource {
 
 		SchemaVersion: 1,
 
-		Schema: AppendBaseAttrSchema(map[string]*schema.Schema{
+		Schema: map[string]*schema.Schema{
 			"application_epg_dn": &schema.Schema{
 				Type:     schema.TypeString,
 				Required: true,
-				ForceNew: true,
 			},
-
 			"contract_name": &schema.Schema{
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: "Use `contract_dn` instead",
+			},
+			"contract_dn": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+				Optional: true,
 			},
```

Comment:
Should we add a CustomizedDiff or similar to check that at least one of contract_name or contract_dn is provided?